README.md

```diff
@@ -109,6 +113,63 @@ constant:
 MyModel = Avromatic::Model.model(schema_name :my_model)
 ```
 
+#### Experimental: Union Support
+
+Avromatic contains experimental support for unions containing more than one
+non-null member type. This feature is experimental because Virtus attributes
+may attempt to coerce between types too aggressively.
+
+In the future, the type coercion used in the gem will be replaced to better
+support the union use case.
+
+For now, if a union contains nested models then it is recommended that you
+assign instances
+
+#### Nested Models
+
+Nested models are models that are embedded within top-level models generated
+using Avromatic. Normally these nested models are automatically generated.
+
+By default, nested models are stored in `Avromatic.nested_models`. This is an
+`Avromatic::ModelRegistry` instance that provides access to previously generated
+nested models by the full name of their Avro schema.
+
+```ruby
+Avromatic.nested_models['com.my_company.test.example']
+#=> <model class>
+```
+
+The `ModelRegistry` can be customized to remove a namespace prefix:
+
+```ruby
+Avromatic.nested_models =
+  Avromatic::ModelRegistry.new(remove_namespace_prefix: 'com.my_company'
+```
+
+By default, top-level generated models reuse `Avromatic.nested_models`. This
+allows nested models to be shared across different generated models.
+A `:nested_models` option can be specified when generating a model. This allows
+the reuse of nested models to be scoped:
+
+```ruby
+Avromatic::Model.model(schema_name, :my_model
+                       nested_models: ModelRegistry.new)
+```
+
+It is also possible to explicitly generate a nested model that should be reused
+and add it to the registry. This is useful when the nested model is extended:
+
+```ruby
+class UsefulSubrecord
+  include Avromatic::Model.build(schema_name: 'useful_subrecord')
+
+  def do_something_custom
+    ...
+  end
+end
+Avromatic.nested_models.register(UsefulSubrecord)
```

Comment:
Is it worth some comments about doing this registration in an initializer so lazy class loading will work?